source/base/tensor_function_parser.cc

```diff
@@ -0,0 +1,375 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2005 - 2019 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+
+#include <deal.II/base/mu_parser_internal.h>
+#include <deal.II/base/patterns.h>
+#include <deal.II/base/tensor.h>
+#include <deal.II/base/tensor_function.h>
+#include <deal.II/base/tensor_function_parser.h>
+#include <deal.II/base/thread_management.h>
+#include <deal.II/base/utilities.h>
+
+#include <boost/random.hpp>
+
+#include <cmath>
+#include <map>
+
+#ifdef DEAL_II_WITH_MUPARSER
+#  include <muParser.h>
+#endif
+
+DEAL_II_NAMESPACE_OPEN
+
+
+template <int rank, int dim, typename Number>
+const std::vector<std::string> &
+TensorFunctionParser<rank, dim, Number>::get_expressions() const
+{
+  return expressions;
+}
+
+
+
+template <int rank, int dim, typename Number>
+TensorFunctionParser<rank, dim, Number>::TensorFunctionParser(
+  const double initial_time)
+  : TensorFunction<rank, dim, Number>(initial_time)
+  , initialized(false)
+  , n_vars(0)
+  , n_components(std::pow(dim, rank))
```

Comment:
```suggestion
  , n_components(Utilities::pow(dim, rank))
```
to avoid the implicit conversion from floating-point number into integer (`clang` warns about that).

Reply:
Done.